packages/cli/src/databases/migrations/postgresdb/1692207852593-MigrateToTimestampZ1692207852593.ts

```diff
@@ -0,0 +1,41 @@
+import type { IrreversibleMigration, MigrationContext } from '@/databases/types';
+
+const defaultTimestampColumns = ['createdAt', 'updatedAt'];
+const tablesWithDefaultTimestamps = [
+	'auth_identity',
+	'credentials_entity',
+	'event_destinations',
+	'installed_packages',
+	'role',
+	'shared_credentials',
+	'shared_workflow',
+	'tag_entity',
+	'user',
+	'workflow_entity',
+];
+
+const additionalColumns = {
+	auth_provider_sync_history: ['endedAt', 'startedAt'],
+	execution_entity: ['startedAt', 'stoppedAt', 'waitTill'],
+	workflow_statistics: ['latestEvent'],
+};
+
+export class MigrateToTimestampZ1692207852593 implements IrreversibleMigration {
+	async up({ queryRunner, tablePrefix }: MigrationContext) {
+		const changeColumnType = async (tableName: string, columnName: string) =>
+			queryRunner.query(
+				`ALTER TABLE "${tablePrefix}${tableName}" ALTER COLUMN "${columnName}" TYPE TIMESTAMPTZ(3)`,
```

Comment:
1. We could use the non-abbreviated version `TIMESTAMP(3) WITH TIME ZONE` to be more explicit.
2. Don't we need to set the `DEFAULT` to mirror `tsColumnOptions`?
3. Should make them non-nullable? But allowing for a few exceptions, e.g. `execution_entity.stoppedAt` and `execution_entity.waitTill`.

Reply:
Should we still set the default so that all the columns have the same precision? Some of them specify it and some don't.

Reply:
don't we need to then make that change across all three DBs?

Reply:
We should! Up to you if that's in scope or not.

Reply:
I think that should be a separate migration. The question is should we add that migration in this PR or not 🤔.

@krynble opinions?

Reply:
I think this can be done in phases, I never dared to tackle this because it has so many possible negative consequences such as displaying wrong data in the app.